components/sup/src/event/nats_message_stream.rs

```diff
@@ -40,57 +43,56 @@ impl NatsMessageStream {
 
-        let mut client = Client::new(vec![url]);
-
-        // Configure the client connect message
-        client.connect_mut()
-              .await
-              .name(format!("hab_client_{}", supervisor_id))
-              .verbose(true)
-              .token(token.to_string());
-
-        // Configure the tls connector
-        let mut tls_connector = TlsConnector::builder();
-        for certificate in habitat_core::tls::native_tls_wrapper::certificates(None)? {
-            tls_connector.add_root_certificate(certificate);
-        }
-        if let Some(certificate) = server_certificate {
-            tls_connector.add_root_certificate(certificate.into());
+        let name = format!("hab_client_{}", supervisor_id); 
+ 
+        //  Not sure how to add a vector into the nats_options object below (in Options::with_token(...))
+        //  when the interface is add_root_certificate
+        let mut hab_certs: Vec<PathBuf> = Vec::new();
+        for certificate in habitat_core::tls::native_tls_wrapper::certificates_pathbuf(None)? {
+            hab_certs.push(certificate);
         }
-        let tls_connector = tls_connector.build()?;
-        client.set_tls_config(tls_connector).await;
+
+        let nats_options  = match server_certificate {
+            Some(nats_options) => {
+                let cert_path: PathBuf = nats_options.into(); 
+                Options::with_token(&token.to_string())
+                    .with_name(&name)
+                    .add_root_certificate(cert_path)
+            }
+            None => {
+                Options::with_token(&token.to_string()) 
+                    .with_name(&name)
+            }
+        };
 
         // Connect to the server. If a timeout was set, we want to ensure we establish a connection
         // before exiting the function. If we do not connect within the timeout we return an error.
         // If we do not have a timeout, we dont care if we can immediately connect. Instead we spawn
         // a future that will resolve when a connection is possible. Once we establish a
         // connection, the client will handle reconnecting if necessary.
-        if let Some(timeout) = connect_method.into() {
-            time::timeout(timeout, client.connect()).await
-                                                    .map_err(|_| Error::ConnectNatsServer)?;
-        } else {
-            let client = Client::clone(&client);
-            tokio::spawn(async move { client.connect().await });
-        }
-
+        
+        let connection : Connection = match connect_method.into() {
+            Some(timeout) => {
+                println!("Timeout used -> {:?}", timeout);
+                time::timeout(timeout,nats_options.connect(&url.to_string()))
+                    .await
+                    .map_err(|_| Error::ConnectNatsServer)??
+            } 
+            None => {
+                println!("Timeout not used");
+                nats_options.connect(&url.to_string())
```

Comment:
looking over the rants documentation it now becomes clear why the former implementation called this in a separate thread. The rants client will continuously try to reconnect if unable to connect or if it becomes disconnected. It looks like the nats library has similar functionality.